examples/material_gallery/lib/demo/tabs_demo.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'widget_demo.dart';
+
+final TabBarSelection _selection = new TabBarSelection();
+final List<String> _iconNames = ["event", "home", "android", "alarm", "face", "language"];
+
+Widget buildTabBar(_) {
+  return new TabBar(
+    selection: _selection,
+    isScrollable: true,
+    labels: _iconNames.map((String iconName) => new TabLabel(text: iconName, icon: "action/$iconName")).toList()
+  );
+}
+
+class TabsDemo extends StatefulComponent {
+  _TabsDemoState createState() => new _TabsDemoState();
+}
+
+class _TabsDemoState extends State<TabsDemo> {
+  double _viewWidth = 100.0;
+
+  void _handleSizeChanged(Size newSize) {
+    setState(() {
+      _viewWidth = newSize.width;
+    });
+  }
+
+  Widget build(_) {
+    return new SizeObserver(
```

Comment:
Yes, I was pretty unhappy having the SizeObserver be the responsibility of the caller.  It didn't report the correct size when I just had the TabBarView incorporate it directly.  I'll try again; sort out exactly what the problem is/was.